specification/redisenterprise/resource-manager/Microsoft.Cache/preview/2024-09-01-preview/redisenterprise.json

```diff
@@ -0,0 +1,2712 @@
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ClusterNameParameter"
+          },
+          {
+            "$ref": "#/parameters/DatabaseNameParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ExportClusterParameters"
+            },
+            "description": "Storage information for exporting into the cluster"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Export operation succeeded."
+          },
+          "202": {
+            "description": "Export operation successfully enqueued; follow the Location header to poll for final outcome.",
+            "headers": {
+              "Location": {
+                "description": "Location URI to poll for result",
+                "type": "string",
+                "format": "uri"
+              },
+              "Azure-AsyncOperation": {
+                "description": "Azure-AsyncOperation URI to poll for result",
+                "type": "string",
+                "format": "uri"
+              }
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Cache/redisEnterprise/{clusterName}/databases/{databaseName}/accessPolicyAssignments/{accessPolicyAssignmentName}": {
+      "put": {
+        "tags": [
+          "RedisEnterprise",
+          "Databases",
+          "AccessPolicyAssignments"
+        ],
+        "operationId": "AccessPolicyAssignment_Create",
+        "description": "Creates access policy assignment for database",
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "original-uri"
+        },
+        "x-ms-examples": {
+          "RedisEnterpriseAccessPolicyAssignmentCreate": {
+            "$ref": "./examples/RedisEnterpriseAccessPolicyAssignmentCreate.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
```

Comment:
>SubscriptionIdParameter [](http://example.com/codeflow?start=95&length=23)

very minor nit, parameter order is not consistent between operations. Pretty sure this won't effect anything except maybe order of parameters in things like generated SDKs etc.

Reply:
@priyankarking  Feel free to re-organize to make consistent.

Reply:
is this comment for new API I have added, does not seems so. Will not make changes here then if it really changes the order of parameters in SDK API.

Reply:
The order for existing APIs (e.g. RedisEnterpriseDatabaseCreate) is:
"databaseName": "default",
"clusterName": "cache1",
"resourceGroupName": "rg1",
"api-version": "2024-09-01-preview",
"subscriptionId": "e7b5a9d2-6b6a-4d2f-9143-20d9a10f5b8f",

For the new AccessPolicyAssignments APIs the order is:
"api-version": "2024-09-01-preview",
"subscriptionId": "e7b5a9d2-6b6a-4d2f-9143-20d9a10f5b8f",
"resourceGroupName": "rg1",
"clusterName": "cache1",
"databaseName": "default"

I don't know if order consistency is important. @TimLovellSmith